node/as/json.js

```diff
@@ -233,12 +233,13 @@ TChannelJSON.prototype.register = function register(
                     'Could not JSON stringify');
             }
 
-            res.headers.as = 'json';
-            res.setOk(respObject.ok);
-            res.send(
-                stringifyResult.value.head,
-                stringifyResult.value.body
-            );
+            if (res.setOk(respObject.ok)) {
```

Comment:
`res.end` isn't sufficient, `res.state` is the source of truth, which is what `res.setOk(...)` is checking in the first place.  In fact I don't trust that we're correctly setting `.end` whenever we move into a terminal `.state`; the `.end` property was always intended as a convenience to record a time, and shouldn't be branched on like a state indicator.